realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -67,7 +67,7 @@
     // Thread pool for all async operations (Query & transaction)
     static final RealmThreadPoolExecutor asyncTaskExecutor = RealmThreadPoolExecutor.newDefaultExecutor();
 
-    final long threadId;
```

Comment:
While I do believe this change is probably safe under the circumstances, it seems to me to increase the fragility of the code.  There is no reason to hold a reference to that gigantic, extremely heavy-weight Thread object.  Unless there is a really good reason for doing this, It seems safer to not to.